force-app/main/default/classes/AffiliationMappingsMapper.cls

```diff
@@ -107,6 +107,56 @@ public virtual with sharing class AffiliationMappingsMapper {
         update afflSettings;
     }
 
+    /*****************************************************************************
+     * @description Updates the auto enrollment mapping on affiliation mapping record
+     * @param mappingName
+     * @param autoProgramEnrollmentStatus
+     * @param autoProgramEnrollmentRole
+     *****************************************************************************/
+    public void updateAutoEnrollmentMappings(
+        String mappingName,
+        String autoProgramEnrollmentStatus,
+        String autoProgramEnrollmentRole
+    ) {
+        Affl_Mappings__c afflSetting = Affl_Mappings__c.getInstance(mappingName);
+        system.debug(afflSetting);
+        afflSetting.Auto_Program_Enrollment_Status__c = autoProgramEnrollmentStatus;
+        afflSetting.Auto_Program_Enrollment_Role__c = autoProgramEnrollmentRole;
+        system.debug(afflSetting);
+        update afflSetting;
+    }
+
+    /*****************************************************************************
+     * @description Copies the status and role from old auto enrollment mapping to
+     * new auto ernollment mapping.
+     * Also, disables auto enrollment for the old auto enrollment mapping
+     * @param mappingName
+     * @param autoProgramEnrollmentStatus
+     * @param autoProgramEnrollmentRole
+     *****************************************************************************/
+    public void copyStatusAndRoleAutoEnrollmentMapping(
```

Comment:
The logic for manipulation might be better served in the service. You can instantiate two models, set the appropriate values, then pass them to be updated.